hdclient.sh

```diff
@@ -0,0 +1,176 @@
+# to another script.
+#
+# The best way to make the functions available is to
+# source in this file from your shell's rc file.
+#
+# To easily configure the access to your Honeydipper
+# daemon, place a file named honeydipper in your home
+# directory under .config, should include following
+# environment variables
+# 
+# export DEFAULT_WEBHOOK_URLPREFIX="< webhook prefix e.g. https://dipper-webhook.myhoneydipper.com >"
+# export DEFAULT_API_URLPREFIX="< api prefix e.g. https://dipper-api.myhoneydipper.com >"
+# export HD_API_TOKEN="< api token >"
+# export HD_WEBHOOK_TOKEN="< webhook token >"
+#
+# examples:
+#
+# $ hdget events
+#
+#     this will list all the events currently executing
+#     workflows in json format.
+#
+# $ hdwebhook mywebhook/test
+#
+#     this will send a webhook request with required
+#     such as https://dipper-webhook.myhoneydipper.com/mywebhook/test
+#     The eventID will be stored in environment variable
+#     HD_EVENT_ID.
+#
+# $ hdwwait
+#
+#     this will wait for the event $HD_EVENT_ID to finish
+#     executing the workflows. check the results with
+#     $HD_SESSION_SUCCESS and $HD_SESSION_FAILURE_ERROR
+#
+#######################################################
+
+
+if [[ -f ~/.config/honeydipper ]]; then
+  source ~/.config/honeydipper
+fi
+
+function hdget() {
+  local api="$1"
+  local urlprefix="${2-$DEFAULT_API_URLPREFIX}"
+
+  if [[ -z "$api" ]]; then
+    echo api not specified >&2
+    return 1
+  fi
+
+  if [[ -z "$urlprefix" ]]; then
+    echo api urlprefix not specified >&2
+    return 1
+  fi
+
+  if [[ -z "$HD_API_TOKEN" ]]; then
+    echo HD_API_TOKEN not specified >&2
+    return 1
+  fi
+
+  # reuse HD_RETURN unless not defined
+  if [[ -z "$HD_RETURN" ]]; then
+    export HD_RETURN="$(mktemp)"
+  fi
+
+  export HD_STATUS_CODE="$(curl -s -o "$HD_RETURN" -w "%{http_code}" -H "Authorization: bearer $HD_API_TOKEN" "$urlprefix/$api")"
+  local ret="$?"
+
+  if [[ -z "$HD_SILENT" ]]; then
+    cat "$HD_RETURN"; echo # add a new line
+  fi
+
+  return "$ret"
+}
+
+function hdwebhook() {
+  local hook="$1"
+  local urlprefix="${2-$DEFAULT_WEBHOOK_URLPREFIX}"
+
+  if [[ -z "$hook" ]]; then
+    echo hook not specified >&2
+    return 1
+  fi
+
+  if [[ -z "$HD_WEBHOOK_TOKEN" ]]; then
+    echo webhook urlprefix not specified >&2
+    return 1
+  fi
+
+  if [[ -z "$HD_WEBHOOK_TOKEN" ]]; then
+    echo HD_WEBHOOK_TOKEN not specified >&2
+    return 1
+  fi
+
+  # reuse HD_WEBHOOK_RETURN unless not defined
+  if [[ -z "$HD_WEBHOOK_RETURN" ]]; then
+    export HD_WEBHOOK_RETURN="$(mktemp)"
+  fi
+
+  export HD_WEBHOOK_STATUS_CODE="$(curl -s -o "$HD_WEBHOOK_RETURN" -w "%{http_code}" -H "Token: $HD_WEBHOOK_TOKEN" "$urlprefix/$hook?accept_uuid")"
+  local ret="$?"
+
+  if [[ "$HD_WEBHOOK_STATUS_CODE" != "200" ]]; then
+    echo got status code "$HD_WEBHOOK_STATUS_CODE" >&2
+    return 1
+  fi
+
+  export HD_EVENT_ID="$(cat "$HD_WEBHOOK_RETURN" | jq -r ".eventID")"
```

Comment:
IMO, this shouldn't be a shell script, but if it is, you need to test for `jq` and `curl` being present.